src/main/java/org/kohsuke/groovy/sandbox/SandboxTransformer.java

```diff
@@ -704,7 +704,7 @@ public void visitExpressionStatement(ExpressionStatement es) {
                     if (mightBePositionalArgumentConstructor((VariableExpression) leftExpression)) {
                         CastExpression ce = new CastExpression(leftExpression.getType(), de.getRightExpression());
                         ce.setCoerce(true);
-                        es.setExpression(transform(new DeclarationExpression(leftExpression, de.getOperation(), transform(ce))));
+                        es.setExpression(transform(new DeclarationExpression(leftExpression, de.getOperation(), ce)));
```

Comment:
and fwiw, when I removed the transformation of the entire `DeclarationExpression`, the new tests failed due to `jenkins.model.Jenkins.getInstance()` not being rejected, so...looks like we're good.